x-pack/plugins/xpack_main/server/lib/feature_registry/feature_registry.ts

```diff
@@ -32,9 +33,50 @@ export interface Feature {
   };
 }
 
+const schema = Joi.object({
+  id: Joi.string()
+    .regex(/^[a-zA-Z0-9_-]+$/)
```

Comment:
I'm fine re-using it within this file, but I'm hesitant to expand it's usage outside of the file because there are likely different reasons for this to change and naming this would be rather hard.